packages/flutter_tools/lib/src/isolated/web_compilation_delegate.dart

```diff
@@ -2,266 +2,257 @@
-      .any((Plugin p) => p.platforms.containsKey(WebPlugin.kConfigKey));
-    final BuildDaemonClient client = await const BuildDaemonCreator().startBuildDaemon(
-      projectDirectory.path,
-      release: mode == BuildMode.release,
-      profile: mode == BuildMode.profile,
-      hasPlugins: hasWebPlugins,
-      initializePlatform: initializePlatform,
-      testTargets: WebTestTargetManifest(
-        testFiles
-          .map<String>((String absolutePath) {
-            final String relativePath = path.relative(absolutePath, from: projectDirectory.path);
-            return '${path.withoutExtension(relativePath)}.*';
-          })
-          .toList(),
-      ),
-    );
-    client.startBuild();
-    bool success = true;
-    await for (final BuildResults results in client.buildResults) {
-      final BuildResult result = results.results.firstWhere((BuildResult result) {
-        return result.target == 'web';
-      }, orElse: () {
-        // Assume build failed if we lack any results.
-        return DefaultBuildResult((DefaultBuildResultBuilder b) => b.status == BuildStatus.failed);
-      });
-      if (result.status == BuildStatus.failed) {
-        success = false;
-        break;
-      }
-      if (result.status == BuildStatus.succeeded) {
-        break;
-      }
+    if (buildInfo.nullSafetyMode == NullSafetyMode.sound) {
+      throwToolExit('flutter test --platform=chrome does not currently support sound mode');
     }
-    if (!success || testOutputDir == null) {
-      return success;
+    final List<String> extraFrontEndOptions = List<String>.of(buildInfo.extraFrontEndOptions ?? <String>[]);
+    if (!extraFrontEndOptions.contains('--no-sound-null-safety')) {
+      extraFrontEndOptions.add('--no-sound-null-safety');
     }
-    final Directory rootDirectory = projectDirectory
-      .childDirectory('.dart_tool')
-      .childDirectory('build')
-      .childDirectory('flutter_web');
+    final Directory outputDirectory = globals.fs.directory(testOutputDir)
+      ..createSync(recursive: true);
+    final List<File> generatedFiles = <File>[];
+    for (final String testFilePath in testFiles) {
+      final List<String> relativeTestSegments = globals.fs.path.split(
+        globals.fs.path.relative(testFilePath, from: projectDirectory.childDirectory('test').path));
+      final File generatedFile = globals.fs.file(
+        globals.fs.path.join(outputDirectory.path, '${relativeTestSegments.join('_')}.test.dart'));
+      generatedFile
+        ..createSync(recursive: true)
+        ..writeAsStringSync(_generateEntrypoint(relativeTestSegments.join('/'), testFilePath));
+      generatedFiles.add(generatedFile);
+    }
+    final StringBuffer buffer = StringBuffer('// @dart=2.8\n');
+    for (final File generatedFile in generatedFiles) {
+      buffer.writeln('import "${globals.fs.path.basename(generatedFile.path)}";');
+    }
+    buffer.writeln('void main() {}');
```

Comment:
Can you please leave a comment explaining why the main.dart file has an empty main function?

Reply:
Done